classes/User.php

```diff
@@ -587,7 +611,7 @@ public function loadData($UPDATE = User::UPDATE_FULL, $remote_view = false): str
         if($this->system->db_last_num_rows != 0) {
             $result = $this->system->db_fetch($result);
             $this->village_location = $result['location'];
-            if($this->location === $this->village_location) {
+            if($this->location->fetchString() === $this->village_location) {
```

Comment:
similar comment as previous - we should move to a comparison format that isn't vulnerable to typecasting errors, like `x:y:z` or a dedicated `equals()` method